iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -533,6 +533,101 @@ void ProcessManager::addPublisherForProcess(const RuntimeName_t& name,
         [&]() { LogWarn() << "Unknown application " << name << " requested a PublisherPort."; });
 }
 
+
+void ProcessManager::addClientForProcess(const RuntimeName_t& name,
+                                         const capro::ServiceDescription& service,
+                                         const popo::ClientOptions& clientOptions,
+                                         const PortConfigInfo& portConfigInfo) noexcept
+{
+    searchForProcessAndThen(
+        name,
+        [&](Process& process) { // create a ClientPort
```

Comment:
I do not know if we had a rule to avoid generic captures? But what do you think about capturing `this`, `&service`, `&serverOptions` and `&name`? But I am not insisting only loud thinking.

This goes also for all other lambdas.

Reply:
Yes, I think we have the rule to explicitly capture. I think the main reason was the ensure we do not capture more than one or two references. For `std::function` the small object optimization should kick in for less than three references but since we now use `cxx::reference_wrapper` I'm not sure if this holds anymore. But I think there is an Autosar rule to be explicit with capturing so I will change it anyway.

Reply:
Since this is legacy code and I already did some cleanup, would you be okay with fixing this when the tools complain and spare some fun for later?